api-flask/app/zonal_stats.py

```diff
@@ -2,24 +2,35 @@
 import logging
 from collections import defaultdict
 from datetime import datetime
+from enum import Enum
 from json import dump, load
+from typing import Any
 from urllib.parse import urlencode
 
-import rasterio
-from app.caching import cache_file, get_json_file
+import rasterio  # type: ignore
+from app.caching import FilePath, cache_file, get_json_file
 from app.timer import timed
-from rasterstats import zonal_stats
-from shapely.geometry import mapping, shape
-from shapely.ops import cascaded_union
-from werkzeug.exceptions import InternalServerError
+from fastapi import HTTPException
+from rasterstats import zonal_stats  # type: ignore
+from shapely.geometry import mapping, shape  # type: ignore
+from shapely.ops import cascaded_union  # type: ignore
 
 logger = logging.getLogger(__name__)
 
 
 DEFAULT_STATS = ["min", "max", "mean", "median"]
 
 
-def get_wfs_response(wfs_params):
+class GroupBy(str, Enum):
```

Comment:
Does this make sense? I was trying to constrain the data in there more than with a mere `str`, but I don't know how many possible values exist for this?

Reply:
Ok, I've switched this to a `NewType` definition, so that we still get strong type validation in the code, but the api will allow any string.